debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -135,6 +139,32 @@ public String version() {
 
         final MySqlEventMetadataProvider metadataProvider = new MySqlEventMetadataProvider();
 
+        Heartbeat heartbeat = null;
+        if (!connectorConfig.getHeartbeatActionQuery().isEmpty()) {
+            heartbeatConnection = new MySqlConnection(new MySqlConnectionConfiguration(config),
+                    connectorConfig.useCursorFetch() ? new MysqlBinaryProtocolFieldReader()
+                            : new MysqlTextProtocolFieldReader());
+
+            Configuration configuration = connectorConfig.getConfig();
+            heartbeat = Heartbeat.create(
+                    configuration.getDuration(Heartbeat.HEARTBEAT_INTERVAL, ChronoUnit.MILLIS),
+                    configuration.getString(DatabaseHeartbeatImpl.HEARTBEAT_ACTION_QUERY),
+                    topicSelector.getHeartbeatTopic(),
+                    connectorConfig.getLogicalName(), heartbeatConnection, exception -> {
+                        String sqlErrorId = exception.getSQLState();
+                        switch (sqlErrorId) {
+                            case "42000":
+                                // error_er_dbaccess_denied_error, see https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_dbaccess_denied_error
+                                throw new DebeziumException("Could not execute heartbeat action (Error: " + sqlErrorId + ")", exception);
+                            case "3D000":
+                                // error_er_no_db_error, see https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_no_db_error
+                                throw new DebeziumException("Could not execute heartbeat action (Error: " + sqlErrorId + ")", exception);
+                            default:
+                                break;
```

Comment:
@gunnarmorling @jpechane  checking back on this..we are really looking to have this enabled for gtid sets replication to downstream processes, this would help us in transaction ordering and check for completeness.

Reply:
Hey @abhishekkh, sorry for the radio silence; lots of things going on at the same time, and this one fell through the cracks a bit. Looking through this now, and going to push some changes myself while doing so. Thanks for hanging in!